scipy/stats/_stats_py.py

```diff
@@ -3191,6 +3206,42 @@ def zmap(scores, compare, axis=0, ddof=0, nan_policy='propagate'):
     array([-1.06066017,  0.        ,  0.35355339,  0.70710678])
 
     """
+    # The docstring explicitly states that it preserves subclasses.
+    # Let's table deprecating that and just get the array API version
+    # working.
+    implementation = _zmap_array_api if SCIPY_ARRAY_API else _zmap_not_array_api
+    return implementation(scores, compare, axis, ddof, nan_policy)
```

Comment:
Up to you but I think it's fairly easy to keep the subclass behaviour by doing `_asarray(..., xp=xp, subok=True)` which might reduce the diff

Reply:
The new implementation is a lot simpler (all the new private functions are used or useful elsewhere). It's also faster for N-d input because it doesn't need `apply_along_axis`. It also warns when there is precision loss (even if a slice is not entirely constant), something we do in `skew` and `kurtosis`. I think it also can be more correct in the edge case of constant `compare`:
```python3
import numpy as np
from scipy import stats
scores = np.arange(10)
compare = np.ones(10)
stats.zscore(compare)  # should be all NaNs
stats.zmap(scores, compare)  # should all be inf 
```
But the original `zmap` produced NaNs in the latter case. (I had that fixed in the new version, but it looks like I messed it up. I'll fix it and add a test case).

So I think it's worth keeping the new implementation rather than trying to convert the old one (which would be more complicated, anyway).

Reply:
Ok sounds good. At what point did you envisage the deprecation of subclasses/removal of the old version?

Reply:
I don't know, but it's the same as everything else that is going to be deprecated in preparation for `SCIPY_ARRAY_API=1` becoming the default and only behavior.